streamz/tests/test_kafka.py

```diff
@@ -141,6 +132,25 @@ def test_from_kafka():
         # absolute sleep here, since we expect output list *not* to change
         yield gen.sleep(1)
         assert out[-1] == b'final message'
+        stream._close_consumer()
```

Comment:
It looks like we do this a bit.  Should we use `weakref.finalize` on the `stream` object to make sure that this happens when the object goes away?